### PR TITLE
cache: Promisify part 3: Read methods

### PR DESCRIPTION
**Pull request by rhansen**
Multiple commits:
* Set DB to null on `close()`
* Use a variable to save some keystrokes
* cache: Promisify `close()`
* cache: Promisify `init()`
* cache: Promisify `get()`
* cache: Promisify `getSub()`
* cache: Promisify `findKeys()`
* tests: Increase `setMax` for `postgres`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -66,9 +66,9 @@ exports.Database.prototype.init = function (callback) {
   const db = new this.dbModule.Database(this.dbSettings);
   this.db = new cacheAndBufferLayer.Database(db, this.wrapperSettings, this.logger);
   if (callback) {
-    this.db.init(callback);
+    util.callbackify(this.db.init.bind(this.db))(callback);
```
Comment (JohnMcLear) [inline index.js:69]: intentional switcheroo?

**Comment by JohnMcLear** [comment]
Oh wow this is cool :)  I looked at doing this but lacked the skill so it's cool to see how it's done!

File: index.js (hunk 1 above)
Comment (rhansen) [reply index.js:69]: Yup. The API for `init()` in `index.js` is unchanged; it's the API for `init()` in `lib/CachAndBufferLayer.js` that's changing with this PR. Thus the need for the switcheroo.

**Comment by JohnMcLear** [comment]
I assume you are working through to solve issues with tests failing?

**Comment by rhansen** [comment]
> I assume you are working through to solve issues with tests failing?

Looks like a flaky test.

**Comment by rhansen** [comment]
> > I assume you are working through to solve issues with tests failing?
> 
> Looks like a flaky test.

Maybe not. I'll have to do some digging. Sigh.

**Comment by Revadike** [comment]
I assume Asyncify is another word for Promisify? If so, I'm very excited for this PR! 
I'm considering using this library for my app, but one of its shortcomings is its callback only support (for now).

**Comment by rhansen** [comment]
I just increased the limit. We need better benchmarking before we can say what a good limit is.

**Comment by rhansen** [comment]
> I assume Asyncify is another word for Promisify?

Yeah, promisify is a better word than asyncify.

> Any ETA?

Probably tomorrow morning.

> I'm considering using this library for my app, but one of its shortcomings is its callback only support (for now).

 This PR only changes internal API. I have some more upcoming changes that will also only change internal API, but eventually we'll be ready to promisify the external API.

**Comment by Revadike** [comment]
I dunno if I should share this, but I was considering to use this for my app, until I discovered ORM's, like typeorm, existed. It's basically this concept, just more established than this.

**Comment by JohnMcLear** [comment]
@Revadike sometimes you need a manual screwdriver, sometimes a powered driver, both have their merits.

**Pull request merged**